src/immutable/ColonyTokenReference.ts

```diff
@@ -0,0 +1,34 @@
+import { $ReadOnly } from 'utility-types';
```

Comment:
Yes, agreed. Their uses seem much more obvious now since their properties have diverged a bit. I also like that `TokenReference`, in the context of users (now `UserTokenReference`) no longer has props that'll _never_ be used within that context (`isNative`, `isExternal`, etc.) 🙂